src/include/txn.i

```diff
@@ -856,21 +854,20 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT
      * the current system.
      */
     if (__wt_txn_visible(session, start.txnid, start.timestamp)) {
-        WT_RET(__wt_update_alloc(session, &buf, &upd, &size, WT_UPDATE_STANDARD));
-        upd->txnid = start.txnid;
-        upd->start_ts = upd->durable_ts = start.timestamp;
-        F_SET(upd, WT_UPDATE_RESTORED_FROM_DISK);
-        *updp = upd;
+        WT_RET(__wt_update_alloc(session, &buf, updp, &size, WT_UPDATE_STANDARD));
+        (*updp)->txnid = start.txnid;
+        (*updp)->start_ts = start.timestamp;
+        F_SET((*updp), WT_UPDATE_RESTORED_FROM_DISK);
         return (0);
     }
 
     /* If there's no visible update in the update chain or ondisk, check the history store file. */
-    WT_ASSERT(session, upd == NULL);
+    WT_ASSERT(session, (*updp) == NULL);
```

Comment:
AFAICT, this assert is literally impossible, so there's no much point in keeping it.

Reply:
Agreed - looks like if the converse is true, we are returning.